test-harness/init.sh

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env bash
```

Comment:
Can you add a documentation here that explains the use of this file? I see what its doing, but its complicated enough that some guidance would help IMO